### DISCUSSION TITLE
Handling 3D data

### DISCUSSION THREAD

**Discussion by NikosMemmos** [TheoMichelot/moveHMM#39] [General]
Hi everyone!

I am new in the area and just started using your software. I am working on 3D data with cells and I was wondering if moveHMM can handle 3D data somehow.

Lastly, I would like to thank you for providing to the community this software! :)

**Comment by TheoMichelot** [comment]
Thanks for the question, unfortunately moveHMM focuses on the analysis of two-dimensional data.

For 3D data, you could take a look at the package momentuHMM, which is an extension of moveHMM. Among other things, momentuHMM allows for any number of observed variables, so you could include that third dimension. You will need to choose what variables to include in the model, based on the states you want to identify. For example, you could choose:

- 3D step length, horizontal turning angle, and vertical turning angle
- 2D step length, horizontal turning angle, and change in z dimension
- change in x dimension, change in y dimension, and change in z dimension

Each set of variables will lead to a different model, probably with different estimated states. You need to derive those variables from the raw x/y/z data yourself, and then you can use them in momentuHMM.

As a starting point, you could for example look at Sections 3.1 and 3.2 of the [momentuHMM vignette](https://cran.r-project.org/web/packages/momentuHMM/vignettes/momentuHMM.pdf). Section 3.1 presents a pretty detailed example of an analysis of 2D data, and Section 3.2 shows a (slightly less detailed) example of an analysis with three observed variables.

**Reply by NikosMemmos** [reply]
Thanks for your reply! My model is based on the length, azimuthal and polar angles. However while the azimuthal angle belongs to (-π, π) range, the polar angle belongs to (-π/2, π/2).  I do not think that using a wrapped Cauchy distribution or Von Mises distribution in the case of the polar angle will cause any issue, but I am wondering what your opinion is on that.

**Reply by TheoMichelot** [reply]
Okay, great, you could combine those three variables into a single HMM in momentuHMM. Using the von Mises or wrapped Cauchy distribution for a variable that's between -π/2 and π/2 will probably lead to some lack-of-fit, but I suspect that it wouldn't be too severe as long as the distributions are centred on zero. You can try it out, and then inspect plots of the estimated distributions (using the plot function in momentuHMM) to assess whether the fit seems okay.
